app/src/org/commcare/views/widgets/LabelWidget.java

```diff
@@ -168,6 +177,14 @@ public LabelWidget(Context context, FormEntryPrompt prompt) {
         addView(questionLayout);
     }
 
+    /**
+     * LabelWidget is used for both ListWidget and ListMultiWidget, and we don't show the clear
+     * button in ListMultiWidget. So this method allows removal of clear button.
+     */
+    public void removeClearButton() {
```

Comment:
Few design suggestions here - 
 1. This should move to `ClearableWidget` so that we localize the logic related to clear button in `ClearableWidget` 
2. An alternative approach to this would be to define a method `showClearButton` in `ClearableWidget` and only add the button if it evals to true. This way we avoid adding the clear button at all for widgets implementing `ClearableWidget` but not wanting to show the button. 
3. A even better approach would be use composition instead of Inheritance here for `ClearableWidget` especially since not all subclasses of `ClearableWidget` ends up showing Clear button in this design.

Reply:
Gotcha, Composition sounds better than inheritance.

Reply:
I have a question here, this `removeClearButton` method is required because `LabelWidget` is used in combination with `ListWidget` and `ListMultiWidget` and we want the clear button only in 1 combination(with `ListWidget`). 
So whatever suggestion I end up using with, I still need this method here I guess.

Reply:
You can inverse the check by introducing another method like `shouldShowClearButton` and have the `ClearableWidget` like - 
````
if(shouldShowClearButton()){
// add clear button
````
The difference is just that we are doing a check while adding the clear button instead of adding and then removing the clear button

Reply:
Ok, I'll offline regarding this.